src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+        for (var {|Rename:i|} = 0; i < array.Count; i++)
+        {
+            var a = array[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task IListOfT()
+        {
+            var text = @"
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new List<int>();
+        foreach [||](var a in list)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            var expected = @"
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new List<int>();
+        for (var {|Rename:i|} = 0; i < list.Count; i++)
+        {
+            var a = list[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task IReadOnlyListOfT()
+        {
+            var text = @"
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new ReadOnly<int>();
+        foreach [||](var a in list)
+        {
+            Console.WriteLine(a);
+        }
+
+    }
+}
+
+class ReadOnly<T> : IReadOnlyList<T>
+{
+    public T this[int index] => throw new System.NotImplementedException();
+    public int Count => throw new System.NotImplementedException();
+    public IEnumerator<T> GetEnumerator() => throw new System.NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new System.NotImplementedException();
+}
+";
+            var expected = @"
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new ReadOnly<int>();
+        for (var {|Rename:i|} = 0; i < list.Count; i++)
+        {
+            var a = list[i];
+            Console.WriteLine(a);
+        }
+
+    }
+}
+
+class ReadOnly<T> : IReadOnlyList<T>
+{
+    public T this[int index] => throw new System.NotImplementedException();
+    public int Count => throw new System.NotImplementedException();
+    public IEnumerator<T> GetEnumerator() => throw new System.NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new System.NotImplementedException();
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task IList()
+        {
+            var text = @"
+using System;
+using System.Collections;
+
+class Test
+{
+    void Method()
+    {
+        var list = new List();
+        foreach [||](var a in list)
+        {
+            Console.WriteLine(a);
+        }
+
+    }
+}
+
+class List : IList
```

Comment:
why did you need these?

Reply:
so, I can test IList case

Reply:
I don't understand.  IList is part of the framework.  Why did you need to duplicate all this code here?

Reply:
because I want to test IList. one in the framework implements multiple interfaces.

Reply:
Sorry, i'm not getting it.  If you want to test IList, then just use IList.  IList doesn't implement anything, it's an interface.  In otherwords, just write it like so:

```c#
+class Test
+{
+    void Method()
+    {
+        var list = default(IList);
+        foreach [||](var a in list)
+        {
+            Console.WriteLine(a);
+        }
+
+    }
+}
```

Reply:
This point still stands.

Reply:
I am fine with the test.

Reply:
I have test where list is interface type as well.